File: taskflow-dotnet-api/src/TaskFlow.Application.Contracts/TodoItems/GetPagedAndFilterTodoItemsRequestDto.cs
using TaskFlow.Application.Contracts.Common;
using TaskFlow.Domain.Shared.Localization;
using TaskFlow.Domain.Shared.TodoItems;
using FluentValidation;
using Microsoft.Extensions.Localization;

namespace TaskFlow.Application.Contracts.TodoItems;

public class GetPagedAndFilterTodoItemsRequestDto : GetListRequestDto
{
    public bool? IsArchived { get; set; }
    public TodoStatus? Status { get; set; }
    public TodoPriority? Priority { get; set; }
    public Guid? CategoryId { get; set; }
}

public class GetPagedAndFilterTodoItemsRequestDtoValidator : AbstractValidator<GetPagedAndFilterTodoItemsRequestDto>
{
    public GetPagedAndFilterTodoItemsRequestDtoValidator(IStringLocalizer<ApplicationResource> localizer)
    {
        Include(new GetListRequestDtoValidator(localizer));
    }
}

File: taskflow-react-client/src/features/todoItems/types/getPagedAndFilterTodoItemsRequestDto.ts
import type { TodoStatus, TodoPriority } from "./todoItemResponseDto";

export interface GetPagedAndFilterTodoItemsRequestDto {
  page: number;
  perPage: number;
  search?: string | null;
  field?: string | null;
  order?: "asc" | "desc" | null;
  isArchived?: boolean | null;
  status?: TodoStatus | null;
  priority?: TodoPriority | null;
  categoryId?: string | null;
}
